spec/board_spec.rb

```diff
@@ -207,4 +207,75 @@
 
     expect(board.position_taken?(8)).to eq(true)
   end
+
+  describe 'full_board?' do
+    it 'returns true when every index in board array is full' do
+      board = Board.new
+
+      board.mark_board(0, 'O')
+      board.mark_board(1)
+      board.mark_board(2, 'O')
+      board.mark_board(3)
+      board.mark_board(4, 'O')
+      board.mark_board(5)
+      board.mark_board(6)
+      board.mark_board(7)
+      board.mark_board(8, 'O')
+
+      expect(board.full_board?).to eq(true)
+    end
+  end
+
+  describe 'full_board?' do
+    it 'returns false when there is an available index in the board array' do
+      board = Board.new
+
+      board.mark_board(0, 'O')
+      board.mark_board(2, 'O')
+      board.mark_board(3)
+      board.mark_board(4, 'O')
+      board.mark_board(6)
+      board.mark_board(7)
+      board.mark_board(8, 'O')
+
+      expect(board.full_board?).to eq(false)
+    end
+  end
+
+  describe 'winning_combination?' do
```

Comment:
You wrote 2 tests for winning combinations (diagonal and vertical) - are you confident that your method works for all winning combinations? Also, it would be helpful if your `it` blocks said what kind of winning combination it was (instead of distinguishing the combinations by the winning mark)

Reply:
What would your suggestion be for the number of combinations? I can be more descriptive in the ` it` block

Reply:
Maybe you should say, returns true when there is a vertical winner, returns true when there is a diagonal winner, returns true when there is a horizontal winner - that kind of thing. And I would have more confidence in the test if you had a board set up where you have 2 Xs in a row somewhere with an O in between, and an O winning somewhere on the board.